### ISSUE TITLE
Unable to use `iterate_sql_insertions` with `enwiki-20220801-page.sql` file.

### ISSUE THREAD

**Issue by brochington** [Erutuon/parse-mediawiki-sql#2]
For some unknown reason, I am getting no output when I try to use `iterate_sql_insertions` with the `enwiki-20220801-page.sql` file found [at the latest wiki dumps](https://dumps.wikimedia.org/enwiki/20220801/). As a sanity check, I've also tested the `category.sql` file, which is work with `iterate_sql_insertions`.

Example:

```rust
let page_sql = unsafe { memory_map("page.sql")? };

// Testing to see if the page.sql file is being read, which it is.
let page_sql_sample = &page_sql[0..5000];
let mut temp_file = File::create("page_sql_sample.sql")?;
temp_file.write_all(page_sql_sample.as_bytes())?;

// Iterate over the first 10 insertions in the page.sql file.
// This should, but doesn't, print the insertions.
iterate_sql_insertions(&page_sql)
  .take(10)
  .for_each(|Page { title, .. }| {
    println!("title: {:?}", title);
  });
```

I'm guessing there is some change to the file that isn't compatible with the iterate function, but I'm not yet sure what. 

for reference sake, here is the beginning bytes of the file:

```sql
-- MySQL dump 10.19  Distrib 10.3.34-MariaDB, for debian-linux-gnu (x86_64)
--
-- Host: db1106    Database: enwiki
-- ------------------------------------------------------
-- Server version	10.4.25-MariaDB-log

/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8mb4 */;
/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
/*!40103 SET TIME_ZONE='+00:00' */;
/*!40014 SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;

--
-- Table structure for table `page`
--

DROP TABLE IF EXISTS `page`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `page` (
  `page_id` int(8) unsigned NOT NULL AUTO_INCREMENT,
  `page_namespace` int(11) NOT NULL DEFAULT 0,
  `page_title` varbinary(255) NOT NULL DEFAULT '',
  `page_is_redirect` tinyint(1) unsigned NOT NULL DEFAULT 0,
  `page_is_new` tinyint(1) unsigned NOT NULL DEFAULT 0,
  `page_random` double unsigned NOT NULL DEFAULT 0,
  `page_touched` binary(14) NOT NULL,
  `page_links_updated` varbinary(14) DEFAULT NULL,
  `page_latest` int(8) unsigned NOT NULL DEFAULT 0,
  `page_len` int(8) unsigned NOT NULL DEFAULT 0,
  `page_content_model` varbinary(32) DEFAULT NULL,
  `page_lang` varbinary(35) DEFAULT NULL,
  PRIMARY KEY (`page_id`),
  UNIQUE KEY `page_name_title` (`page_namespace`,`page_title`),
  KEY `page_random` (`page_random`),
  KEY `page_len` (`page_len`),
  KEY `page_redirect_namespace_len` (`page_is_redirect`,`page_namespace`,`page_len`)
) ENGINE=InnoDB AUTO_INCREMENT=71432794 DEFAULT CHARSET=binary ROW_FORMAT=COMPRESSED;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `page`
--

/*!40000 ALTER TABLE `page` DISABLE KEYS */;
INSERT INTO `page` VALUES (10,0,'AccessibleComputing',1,0,0.33167112649574004,'20220721184125','20220721184148',1002250816,111,'wikitext',NULL)
```

**Comment by brochington** [comment]
Update: I forked this repo and linked against a local copy of the codebase, which looks to work. Perhaps the latest code hasn't been pushed to crates.io?

**Comment by Erutuon** [comment]
Yep, I removed the `page_restrictions` field two months ago to fix parsing of `page.sql`, but didn't publish it. Published now as version 0.9.0. It should work for you now.

**Comment by brochington** [comment]
Ah, wonderful. 0.9.0 is working for me. Thank you for the quick response!
